ensight_lib/include/ensightasciireader.h

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c) 2016 Fraunhofer ITWM
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+
+#ifndef ENSIGHTASCIIREADER_H
+#define ENSIGHTASCIIREADER_H
+
+#include <string>
+
+#include "../include/ensightdef.h"
+
+class EnsightObj;
+class QFile;
+class QString;
+
+
+namespace Ensight
+{
+namespace Reader
+{
+namespace detail
+{
+
+/**
+ * @brief Read an Ensight Geometry file from filename.
+ * @param ensight The ensight file the loaded geometry is added to
+ * @param filename The filename of the Geometry file
+ * @param timestep The current timestep in the Ensight object
```

Comment:
For data sets containing multiple time steps, `readTimeStep` either specifies a single time step to be read or -1 if all time steps are to be read. The current `timestep`, on the other hand, is used for incrementally reading several time steps to keep track which steps have already been read.
I.e. one parameter specifies "what have we read so far", the other specifies "which parts are we interested in".

However, since these are internal functions, the details are not relevant to the user.